iree/turbine/kernel/wave/index_sequence_analysis.py

```diff
@@ -124,6 +129,7 @@ def set_node_indices(trace: CapturedTrace, constraints: list[Constraint]):
     mma_index, mma_slices = get_mma_dimensional_mapping(
         trace, get_hardware_constraint(constraints)
     )
+    trace.walk(partial(set_vector_shapes, constraints, mma_index, mma_slices))
```

Comment:
No Action required: in the future perhaps we can do `set_vector_shapes` and `set_node_index` in a single walk. (still keeping the two functions apart though)

Reply:
good point. will keep that for a future PR.